ansible_ai_connect/main/ssl_manager.py

```diff
@@ -0,0 +1,181 @@
+        self.service_ca_path = os.environ.get("SERVICE_CA_PATH")
+        self.requests_ca_bundle = os.environ.get("REQUESTS_CA_BUNDLE")
+
+        # Legacy settings support for backward compatibility
+        self._service_ca_path = getattr(settings, "SERVICE_CA_PATH", None)
+
+        logger.info("SSL Manager: Infrastructure mode initialized")
+        logger.debug("SSL Manager: Combined CA bundle: %s", self.combined_ca_bundle)
+        logger.debug("SSL Manager: Service CA: %s", self.service_ca_path)
+        logger.debug("SSL Manager: Requests CA bundle: %s", self.requests_ca_bundle)
+        logger.debug("SSL Manager: Legacy service CA: %s", self._service_ca_path)
+
+    def _get_ca_bundle_path(self) -> Optional[str]:
+        """Get the best available CA bundle path with prioritized fallback.
+        Priority order:
+        1. Infrastructure-provided combined bundle (operator creates this)
+        2. REQUESTS_CA_BUNDLE environment variable
+        3. Service CA from environment
+        4. Legacy service CA from settings
+        5. None (use system defaults)
+        Returns:
+            Path to CA bundle file or None for system defaults
+        """
+        # Priority 1: Infrastructure-provided combined bundle (best)
+        if self.combined_ca_bundle and os.path.exists(self.combined_ca_bundle):
+            logger.debug(
+                "SSL Manager: Using infrastructure combined CA bundle: %s", self.combined_ca_bundle
+            )
+            return self.combined_ca_bundle
+
+        # Priority 2: Environment REQUESTS_CA_BUNDLE
+        if self.requests_ca_bundle and os.path.exists(self.requests_ca_bundle):
+            logger.debug("SSL Manager: Using REQUESTS_CA_BUNDLE: %s", self.requests_ca_bundle)
+            return self.requests_ca_bundle
+
+        # Priority 3: Service CA (for internal services)
+        if self.service_ca_path and os.path.exists(self.service_ca_path):
+            logger.debug("SSL Manager: Using service CA: %s", self.service_ca_path)
+            return self.service_ca_path
+
+        # Priority 4: Legacy settings service CA
+        if self._service_ca_path and os.path.exists(self._service_ca_path):
+            logger.debug("SSL Manager: Using legacy service CA: %s", self._service_ca_path)
+            return self._service_ca_path
+
+        # No custom CA bundle found - use system defaults
+        logger.debug("SSL Manager: No custom CA bundle found, using system defaults")
+        return None
+
+    def get_requests_session(self, verify_ssl: bool = True) -> requests.Session:
+        """Get requests session with infrastructure-managed SSL configuration.
+        Args:
+            verify_ssl: Whether SSL verification should be enabled
+        Returns:
+            Configured requests session with proper SSL settings.
+            Falls back to system defaults if custom CA bundles are not accessible.
+        """
+        session = requests.Session()
+
```

Comment:
Moving the following block at the top of the function will reduce the complexity, like you did here https://github.com/ansible/ansible-ai-connect-service/pull/1756/files#diff-5383513b0ec56a18282de119eaa5dfb5eda4512a099be14316686c5f365989eaR142:

```python
if not verify_ssl:
    session.verify = False
    logger.info("SSL Manager: SSL verification disabled")
    return session
```

This way you can remove one level of indentation for the rest of the logic.